### ISSUE TITLE
sns.heatmap top and bottom boxes are cut off

### ISSUE THREAD

**Issue by ResidentMario** [mwaskom/seaborn#1773]
With `seaborn 0.9.0` and `matplotlib 3.1.1`, the topmost and bottommost row of boxes in a `seaborn` plot are partially cut off:

```python
import seaborn as sns
import numpy as np

np.random.seed(42)
sns.heatmap(np.random.random((10, 10)))
```

![heatmap](https://user-images.githubusercontent.com/3466341/60834654-24204280-a18f-11e9-9212-b792c4f1a7a4.png)

As another example, consider this [demo from the gallery](https://seaborn.pydata.org/examples/heatmap_annotation.html):

```
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

# Load the example flights dataset and conver to long-form
flights_long = sns.load_dataset("flights")
flights = flights_long.pivot("month", "year", "passengers")

# Draw a heatmap with the numeric values in each cell
f, ax = plt.subplots(figsize=(9, 6))
sns.heatmap(flights, annot=True, fmt="d", linewidths=.5, ax=ax)
```

![heatmap](https://user-images.githubusercontent.com/3466341/60834759-6a75a180-a18f-11e9-81fc-8dc659e4f0e6.png)

This is a pretty major visual issue. Apologies if this has been reported before, but I couldn't find a record of such in the GH issues.

**Comment by ResidentMario** [comment]
[This was a `matplotlib` regression introduced in 3.1.1 which has been fixed in 3.1.2 (still forthcoming)](https://gitter.im/matplotlib/matplotlib?at=5d239514f5dd1457424d7b09). For now the fix is to downgrade `matplotlib` to a prior version.

**Comment by beverly0005** [comment]
solution also at [#14751](https://github.com/matplotlib/matplotlib/issues/14751)

**Comment by SalMac86** [comment]
Here's what I figured out as a work-around:
```
# fix for mpl bug that cuts off top/bottom of seaborn viz
b, t = plt.ylim() # discover the values for bottom and top
b += 0.5 # Add 0.5 to the bottom
t -= 0.5 # Subtract 0.5 from the top
plt.ylim(b, t) # update the ylim(bottom, top) values
plt.show() # ta-da!
```

**Comment by MaozGelbart** [comment]
> [This was a `matplotlib` regression introduced in 3.1.1 which has been fixed in 3.1.2 (still forthcoming)](https://gitter.im/matplotlib/matplotlib?at=5d239514f5dd1457424d7b09). For now the fix is to downgrade `matplotlib` to a prior version.

Matplotlib 3.1.2 has been [released](https://github.com/matplotlib/matplotlib/releases/tag/v3.1.2) (also available for conda users through conda-forge using ``conda install -c conda-forge matplotlib=3.1.2``). This fixes the issue.

**Comment by kkairu** [comment]
> [This was a `matplotlib` regression introduced in 3.1.1 which has been fixed in 3.1.2 (still forthcoming)](https://gitter.im/matplotlib/matplotlib?at=5d239514f5dd1457424d7b09). For now the fix is to downgrade `matplotlib` to a prior version.

Upgrade to Matplotlib 3.1.2 sorts the issue
